eth/stagedsync/stage_interhashes.go

```diff
@@ -82,8 +82,9 @@ func SpawnIntermediateHashesStage(s *StageState, u Unwinder, tx kv.RwTx, cfg Tri
 
 	var expectedRootHash common.Hash
 	var headerHash common.Hash
-	var syncHeadHeader *types.Header
+	var parentHeader common.Hash
```

Comment:
I feel - this way cause more dangerous bugs: 
- now - if not set variable - it immediately fails with "nil pointer"
- but in this case if not set the variable - it will report block with zero-hash - and maybe fail somewhere in far future - and it will hard to debug.


Also this PR doesn't actually change anything (in logic). If by some reason now variable syncHeadHeader not set, then variable parentHeader will not set either.

Reply:
Yes, this PR doesn't actually change anything but avoid panic crash. 

The root cause of panic is the `syncHeadHeader` pointer will be released in advance.

Reply:
"Yes, this PR doesn't actually change anything but avoid panic crash." - yes, but it's not enough. Replace nil by zero-hash will not cause panic, but may lead to other "fail later" bug - because likely `ReportBadHeaderPoS` doesn't expect zero-hash. 

"The root cause of panic is the syncHeadHeader pointer will be released in advance." - no, i don't believe in it - golang doesn't behave so undetermenistic. If variable is used later - it will not be free earlier. Let's look at next pattern: 
```
var a *int
var b int
if condition {
     a = &b
}

if anotherCondition {
    ReportBadHeaderPoS(a)
}
```
a is nil sometime because `condition` and `anotherCondition` are different conditions - not because golang is stupid and released a before ReportBadHeaderPoS(a).

Reply:
Then you can do `if cfg.hd != nil && syncHeadHeader != nil`

Reply:
what's it means?

Reply:
before reporting syncHeadHeader.ParentHash as a bad block hash you check wether syncHeadHeader is nil

Reply:
> The `condition` and `anotherCondition` both have `cfg.checkRoot`. I can't know where and when would cause the `syncHeadHeader` to be nil but the `*types.Header` was released in advance.

then I think - nothing to fix. i think nobody can reproduce this error. 
if some other user can - I bet they didn't update/recompile erigon. Change line number of ReportBadHeaderPoS() - you will see if such user give you new stack trace or old one. 
"was released in advance." - i'm sure Go doesn't release pointers "in advance" - you can try to proof it by `runtime.SetFinalizer` on this object.

Reply:
But it really works in my test. Without it, erigon will get a panic when write wrong trie root.

Reply:
> before reporting syncHeadHeader.ParentHash as a bad block hash you check wether syncHeadHeader is nil

OK, I'll try.

Reply:
> But it really works in my test. Without it, erigon will get a panic when write wrong trie root.

"erigon will get a panic when write wrong trie root." - ok. but this PR doesn't change logic, instead of nil pointer it will report zero-hash as bad block. As I understand it's also wrong behavior - instead of reporting block with correct hash, you reporting block with zero-hash. Ok, no panic, but Bug/WrongLogic is still there. 

Can I reproduce?

Reply:
Ok, I got it.